### DISCUSSION TITLE
Why doesn't it rerender after I send a cross-component action?

### DISCUSSION THREAD

**Discussion by MatherLyn** [statelyai/xstate#1742] [Q&A] [answered]
First, I want to express my appreciate on the use of state design pattern in xstate.
Below is my dilemma:

I use a button to control the state of a component.
And I create two machines, one is for the button, the other is for the component.
In the button's machine, I define the onclick event:
![image](https://user-images.githubusercontent.com/48438633/102579285-27b34980-4137-11eb-831f-8d62681e9bcc.png)
And this is how the action is defined:
![image](https://user-images.githubusercontent.com/48438633/102579333-444f8180-4137-11eb-84b0-d4714f1dd9f2.png)
And this is the button's tsx:
![image](https://user-images.githubusercontent.com/48438633/102579657-deafc500-4137-11eb-8ba1-b7beef13d914.png)

In the component's machine, I define the toggle event:
![image](https://user-images.githubusercontent.com/48438633/102579473-7b259780-4137-11eb-92c8-a047b011dc05.png)
actions:
![image](https://user-images.githubusercontent.com/48438633/102579725-0010b100-4138-11eb-8960-22fa0de4d496.png)
tsx:
![image](https://user-images.githubusercontent.com/48438633/102579759-17e83500-4138-11eb-9fca-aa20bf62aaed.png)

When I click the button, I found the style in the component doesn't change. I added breakpoint and found the action was successfully sent, but the state in the component's machine did not update, which means the event was not be emitted.

Thanks for reading, and please give me a hand.

**Comment by davidkpiano** [comment, marked as answer]
Thanks for the screenshots, but a [CodeSandbox (React template)](https://codesandbox.io/s/xstate-react-template-3t2tg) is necessary in order to debug this. Can you please make a minimal reproduction?

**Reply by MatherLyn** [reply]
Thanks for replying. I found the problem that I did not share the same instance of the interpreter between these two component. After I learned the actor model, I spawn the machines and mount them on a root machine and solve this problem~
